Source/Tools/FEXServer/CoreDumpService.cpp

```diff
@@ -0,0 +1,365 @@
+#include "CoreDumpService.h"
+#include "ProcessPipe.h"
+#include "Common/FEXServerClient.h"
+
+#include "Common/SocketUtil.h"
+
+#include <FEXCore/Utils/LogManager.h>
+
+#include <dirent.h>
+#include <mutex>
+#include <poll.h>
+#include <pwd.h>
+#include <span>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <thread>
+#include <vector>
+
+#include <fmt/chrono.h>
+
+namespace CoreDumpService {
+  void CoreDumpClass::ConsumeFileMapsFD(int FD) {
+    lseek(FD, 0, SEEK_SET);
+    DIR *dir = fdopendir(dup(FD));
+    struct dirent *entry;
+    char Tmp[PATH_MAX];
+    FileMapping::FileMapping *CurrentMap{};
+    while ((entry = readdir(dir)) != nullptr) {
+      struct stat buf{};
+      if (fstatat(FD, entry->d_name, &buf, 0)) {
+        int LinkSize = readlinkat(FD, entry->d_name, Tmp, 512);
+        if (LinkSize > 0) {
+          auto ReadlinkView = std::string_view(Tmp, LinkSize);
+          uint64_t Begin, End;
+          if (sscanf(entry->d_name, "%lx-%lx", &Begin, &End) == 2) {
+            if (CurrentMap && CurrentMap->Path == ReadlinkView) {
+              // Extending file mapping
+              CurrentMap->End = End;
+            }
+            else {
+              // Start a new file mapping
+              CurrentMap = &FileMappings.emplace_back(FileMapping::FileMapping{Begin, End, fextl::string(ReadlinkView)});
```

Comment:
Using `push_back` has no performance overhead for temporaries and allows this to be less chatty:
```suggestion
              CurrentMap = &FileMappings.push_back({Begin, End, fextl::string(ReadlinkView)});
```
Maybe even this:
```suggestion
              CurrentMap = &FileMappings.push_back({Begin, End, { ReadlinkView }});
```

Reply:
std::list doesn't return a reference to the object pushed in this case.